tests/Feature/ModulesTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Module;
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class ModulesTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    function modules_can_be_seen()
+    {
+        $standardModule = factory(Module::class)->create([
+            'name' => 'Standard module',
+            'is_bonus' => 0,
+        ]);
+
+        $bonusModule = factory(Module::class)->create([
+            'name' => 'Bonus module',
+            'is_bonus' => 1,
+        ]);
+
+        $response = $this->get("/en/modules");
+
+        $response->assertSeeInOrder([
+            $standardModule->name,
+            $bonusModule->name,
+        ]);
+    }
+
+    /** @test */
+    function list_of_standard_modules_only_contains_standard_modules()
+    {
+        $standardModule = factory(Module::class)->create([
+            'name' => 'Standard module',
+            'is_bonus' => 0,
+        ]);
+
+        $bonusModule = factory(Module::class)->create([
+            'name' => 'Bonus module',
+            'is_bonus' => 1,
+        ]);
+
+        $response = $this->get("/en/modules");
+
+        $this->assertTrue(count($this->getResponseData($response, 'standardModules')) == 1);
+        $this->assertTrue($this->getResponseData($response, 'standardModules')[0]->name == $standardModule->name);
+        $this->assertFalse($this->getResponseData($response, 'standardModules')[0]->name == $bonusModule->name);
+    }
+
+    /** @test */
+    function list_of_bonus_modules_only_contains_bonus_modules()
+    {
+        $standardModule = factory(Module::class)->create([
+            'name' => 'Standard module',
+            'is_bonus' => 0,
+        ]);
+
+        $bonusModule = factory(Module::class)->create([
+            'name' => 'Bonus module',
+            'is_bonus' => 1,
+        ]);
+
+        $response = $this->get("/en/modules");
+
+        $this->assertTrue(count($this->getResponseData($response, 'bonusModules')) == 1);
```

Comment:
I think you can replace these `getResponseData` assertions with `assertViewHas`: https://laravel.com/docs/6.x/http-tests#assert-view-has

Reply:
A closure does the trick. I'll update it and send you a message here when it's done.

Reply:
[Matt](@mattstauffer), I've updated the assertions. Is this what you had in mind?

Reply:
@rontu85 hm, i'd be curious to debug that `getKey` error. I'll take a look as soon as i can get a chance!

Reply:
@mattstauffer no need to debug this. It just tries to compare an eloquent model with a collection which does not work. The method getKey should come from the model. As stated above a closure was necessary to solve this.